docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,300 @@
+:toc: macro
+
+= RFC 2: tBTCv2 Group Selection and Key Generation
+
+:icons: font
+:numbered:
+toc::[]
+
+== Overview
+
+The key generation uses the protocol from `tss-lib`.
+
+We select members from the sortition pool,
+and they perform one or more attempts to complete the key generation.
+
+Inactive members are removed from the DKG participants
+and the resulting wallet if the DKG is successful.
+
+If there is misbehaviour,
+the accusations will be resolved by governance
+as the protocol in `tss-lib` is not attributable.
+
+=== Parameter selection
+
+For tbtc2 wallets,
+we need to choose the group size _N_
+(how many members we select from the sortition pool)
+and the threshold _T_
+(how many members must cooperate to generate a threshold ECDSA signature.
+
+In addition, we need to choose the critical size _C_
+which is number of members required for a successful key generation.
+Some of the selected _N_ members may not be available for the key generation.
+To reduce sortition costs,
+we let some of the original _N_ be inactive
+as long as at least _C_ participants can be assembled to complete the DKG.
+After successful DKG completion,
+the created wallet will have a number of members
+that is between _C_ and _N_ (inclusive).
+The inactive members that did not participate in the successful DKG
+are removed from the wallet's members and are ineligible for rewards.
```

Comment:
Modifying sortition pools to accommodate constant rewards would be quite a bit of work but it can be done. The penalty for inactive members would then come from being removed from the pool and having to rejoin and wait the initialisation period before receiving rewards.

Rewarding the members of a wallet when the wallet sweeps deposits/executes redemptions would be simpler and we could use similar logic as in the random beacon to add extra reward for transaction submitters, but it makes the rewards for any given operator a lot more random. A hybrid approach (some reward to all, some to members) is also possible.